### ISSUE TITLE
include ignoreErrors attribute not working for non-existent file

### ISSUE THREAD

**Issue by ghills** [NLog/NLog#2094]
**Type** (choose one): Bug

**NLog version**: 4.4.8

**Platform**: .Net 4.6.1

**Current NLog config** (xml or C#, if relevant)

```xml
<?xml version="1.0" encoding="utf-8" ?>
<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
      xsi:schemaLocation="http://www.nlog-project.org/schemas/NLog.xsd NLog.xsd"
      autoReload="true"
      throwExceptions="true"
      internalLogLevel="Off" internalLogFile="c:\temp\nlog-internal.log">
  
  <include file="DeveloperConfig\NLog.config" ignoreErrors="true" />

  <!-- optional, add some variables
  https://github.com/nlog/NLog/wiki/Configuration-file#variables
  -->
  <variable name="myvar" value="myvalue"/>

  <!--
  See https://github.com/nlog/nlog/wiki/Configuration-file
  for information on customizing logging rules and outputs.
   -->
  <targets async="true">

    <!--
    add your targets here
    See https://github.com/nlog/NLog/wiki/Targets for possible targets.
    See https://github.com/nlog/NLog/wiki/Layout-Renderers for the possible layout renderers.
    -->

    <!--
    Write events to a file with the date in the filename.
    <target xsi:type="File" name="f" fileName="${basedir}/logs/${shortdate}.log"
            layout="${longdate} ${uppercase:${level}} ${message}" />
    -->
    <target xsi:type="Trace" name="trace" />
  </targets>

  <rules>
    <!-- add your logging rules here -->

    <!--
    Write all events with minimal level of Debug (So Debug, Info, Warn, Error and Fatal, but not Trace)  to "f"
    <logger name="*" minlevel="Debug" writeTo="f" />
    -->
    <logger name="*" minlevel="Info" writeTo="trace"/>
  </rules>
</nlog>

```

- **What is the current result?** If I enable NLog exceptions it is hitting an exception when trying to include a non-existent file.
- **What is the expected result?** Because I have ignoreErrors="true" for the include, my understanding is it should try and include the file but ignore it if the file doesn't exist.
- **Did you checked the [Internal log](https://github.com/NLog/NLog/wiki/Internal-Logging)?**
```
2017-05-05 08:42:55.7603 Debug ScanAssembly('NLog, Version=4.0.0.0, Culture=neutral, PublicKeyToken=5120e14c03d0593c')
2017-05-05 08:42:55.7843 Debug Start auto loading, location: C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\bin
2017-05-05 08:42:55.7853 Info Auto loading assembly file: C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\bin\NLog.Targets.Syslog.dll
2017-05-05 08:42:55.7853 Info NLog.Targets.Syslog, Version=4.0.0.0, Culture=neutral, PublicKeyToken=a15d6ba50060b73f. File version: 4.0.0.0. Product version: 4.0.0.
2017-05-05 08:42:55.7853 Debug ScanAssembly('NLog.Targets.Syslog, Version=4.0.0.0, Culture=neutral, PublicKeyToken=a15d6ba50060b73f')
2017-05-05 08:42:55.7853 Info Auto loading assembly file: C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\bin\NLog.Targets.Syslog.dll succeeded!
2017-05-05 08:42:55.7853 Debug Auto loading done
2017-05-05 08:42:55.8073 Trace FindReachableObject<System.Object>:
2017-05-05 08:42:55.8073 Trace Scanning SimpleLayout ''DeveloperConfig\NLog.config''
2017-05-05 08:42:55.8073 Trace  Scanning LiteralLayoutRenderer 'Layout Renderer: ${literal}'
2017-05-05 08:42:55.8243 Error Error when including 'DeveloperConfig\NLog.config'. Exception: System.IO.FileNotFoundException: Included file not found: C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\DeveloperConfig\NLog.config
   at NLog.Config.XmlLoggingConfiguration.ParseIncludeElement(NLogXmlElement includeElement, String baseDirectory, Boolean autoReloadDefault)
2017-05-05 08:42:55.8243 Error Parsing configuration from C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\NLog.config failed. Exception: NLog.NLogConfigurationException: Exception when parsing C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\NLog.config.  ---> System.IO.FileNotFoundException: Included file not found: C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\DeveloperConfig\NLog.config
   at NLog.Config.XmlLoggingConfiguration.ParseIncludeElement(NLogXmlElement includeElement, String baseDirectory, Boolean autoReloadDefault)
   at NLog.Config.XmlLoggingConfiguration.ParseNLogElement(NLogXmlElement nlogElement, String filePath, Boolean autoReloadDefault)
   at NLog.Config.XmlLoggingConfiguration.ParseTopLevel(NLogXmlElement content, String filePath, Boolean autoReloadDefault)
   at NLog.Config.XmlLoggingConfiguration.Initialize(XmlReader reader, String fileName, Boolean ignoreErrors)
   --- End of inner exception stack trace ---
```
- **Please post full exception details (message, stacktrace, inner exceptions)**
```
InnerException: 
       HResult=-2146233088
       Message=Exception when parsing C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\NLog.config. 
       Source=NLog
       StackTrace:
            at NLog.Config.XmlLoggingConfiguration.Initialize(XmlReader reader, String fileName, Boolean ignoreErrors)
            at NLog.Config.XmlLoggingConfiguration..ctor(String fileName, Boolean ignoreErrors, LogFactory logFactory)
            at NLog.LogFactory.LoadLoggingConfiguration(String configFile)
            at NLog.LogFactory.get_Configuration()
            at NLog.LogFactory.GetLogger(LoggerCacheKey cacheKey)
            at NLog.LogFactory.GetLogger(String name)
            at NLog.LogManager.GetCurrentClassLogger()
            at AgencyRM.Tracker.Data.Migrations.MigrationHelper..cctor() in C:\dev\AgencyRM.Tracker\AgencyRM.Tracker.Data\Migrations\MigrationHelper.cs:line 11
       InnerException: 
            HResult=-2147024894
            Message=Included file not found: C:\dev\AgencyRM.Tracker\AgencyRM.Tracker\DeveloperConfig\NLog.config
            Source=NLog
            StackTrace:
                 at NLog.Config.XmlLoggingConfiguration.ParseIncludeElement(NLogXmlElement includeElement, String baseDirectory, Boolean autoReloadDefault)
                 at NLog.Config.XmlLoggingConfiguration.ParseNLogElement(NLogXmlElement nlogElement, String filePath, Boolean autoReloadDefault)
                 at NLog.Config.XmlLoggingConfiguration.ParseTopLevel(NLogXmlElement content, String filePath, Boolean autoReloadDefault)
                 at NLog.Config.XmlLoggingConfiguration.Initialize(XmlReader reader, String fileName, Boolean ignoreErrors)
            InnerException: 
```
- **Are there any work arrounds?** no
- **Is there a version in which it did worked?** unknown

**Comment by 304NotModified** [comment]
could you try with `<nlog  throwExceptions="false"`?

**Comment by 304NotModified** [comment]
the bug is here:

https://github.com/NLog/NLog/blob/e37aca88fb569fe490b509e3873a928145309c98/src/NLog/Config/XmlLoggingConfiguration.cs#L1112

**Comment by ghills** [comment]
I turned off the exceptions and added a file output target and the application started up fine and the NLog output was properly being captured. So it looks like it will ignore the error enough to keep going and use the configuration but it doesn't suppress the exception?

**Comment by 304NotModified** [comment]
well the exceptions are caught when throwExceptions=false.

This is indeed a bug, but the workaround is to use throwExceptions=false.

**Comment by 304NotModified** [comment]
we need to unit test this case, and fix it :)

**Comment by 304NotModified** [comment]
could you help with that? (writing unit test)

**Comment by ghills** [comment]
I'll give it a go

**Comment by 304NotModified** [comment]
> @304NotModified do you want me to just add the unit test or also modify the code to resolve the issue?

That was also allowed :) Any way, I think it fixed it. Pushed it to your branch. Will deploy soon :)

**Comment by 304NotModified** [comment]
Fixed in #2095 :)

**Comment by 304NotModified** [comment]
NLog 4.4.9 is in the deploy pipeline! Will be online in an hour!
